src/opnsense/mvc/app/controllers/OPNsense/Dnsmasq/Api/SettingsController.php

```diff
@@ -50,7 +68,21 @@ public function getAction()
     /* hosts */
     public function searchHostAction()
     {
-        return $this->searchBase('hosts');
+        $filters = $this->request->get('tags');
+        $tagMap = $this->getTagUuidMap();
+
+        $filter_funct = function ($record) use ($filters, $tagMap) {
```

Comment:
function could be generalised, the fields to choose from are `set_tag`, `tag` and `interface` when they exist.